docs.md

```diff
@@ -276,3 +276,26 @@ If `roletemplates.builtin` is true then all fields are immutable except:
  ### Deletion check
 
 RoleTemplate can not be deleted if they are referenced by other RoleTemplates via `roletemplates.roleTemplateNames` or by GlobalRoles via `globalRoles.inheritedClusterRoles`
+
+# provisioning.cattle.io/v1 
+
+## Cluster 
+
+### Validation Checks
+
+#### cluster.spec.clusterAgentDeploymentCustomization and cluster.spec.fleetAgentDeploymentCustomization
+
+`Key rule`: The keys are being validated to match the kubernetes requirement. 63 Characteres maximum, Regex:
```

Comment:
Odd wording, maybe `The keys being validated against the kubernetes requirements`, or something along those lines?